src/effects/specialeffectchainslots.cpp

```diff
@@ -71,6 +77,10 @@ PerGroupEffectChainSlot::PerGroupEffectChainSlot(const QString& group,
         : EffectChainSlot(chainSlotGroup, iChainNumber, pEffectsManager,
                           SignalProcessingStage::Prefader, false,
                           chainSlotGroup) {
+    setup(group);
+}
+
+void PerGroupEffectChainSlot::setup(const QString& group) {
```

Comment:
Why did you split this out from the constructor?

Reply:
We cannot call virtual functions inside the constructor/destructor. It returns an error if we try to do the same. This is the same reason I'm passing the chainSlotGroup and the group both to the base class PerGroupEffectChainSlot.

Reference: https://stackoverflow.com/questions/962132/calling-virtual-functions-inside-constructors

Reply:
I think this won't be a problem if we move the call to `addEffectSlot` into the EqualizerEffectChainSlot and QuickEffectChainSlot constructors instead of the base PerGroupEffectChainSlot constructor. That will also allow adding more effect slots for the QuickEffects than the EQs.